web/modules/custom/joinup_subscription/src/Form/UnsubscribeFromAllCollectionsForm.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\joinup_subscription\Form;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\ConfirmFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Renderer;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Drupal\Core\Session\AccountInterface;
+use Drupal\Core\Url;
+use Drupal\og\MembershipManagerInterface;
+use Drupal\og\OgMembershipInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Form that allows the user to unsubscribe from all groups.
+ */
+class UnsubscribeFromAllCollectionsForm extends ConfirmFormBase {
+
+  /**
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The membership manager service.
+   *
+   * @var \Drupal\og\MembershipManagerInterface
+   */
+  protected $membershipManager;
+
+  /**
+   * The renderer service.
+   *
+   * @var \Drupal\Core\Render\Renderer
+   */
+  protected $renderer;
+
+  /**
+   * The user entity from the route.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $user;
+
+  /**
+   * Constructs a UnsubscribeFromAllCollectionsForm.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\og\MembershipManagerInterface $membership_manager
+   *   The membership manager service.
+   * @param \Drupal\Core\Render\Renderer $renderer
+   *   The renderer service.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, MembershipManagerInterface $membership_manager, Renderer $renderer, RouteMatchInterface $route_match) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->membershipManager = $membership_manager;
+    $this->renderer = $renderer;
+    $this->user = $route_match->getParameter('user');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('og.membership_manager'),
+      $container->get('renderer'),
+      $container->get('current_route_match')
```

Comment:
We actually don't need to retrieve the route matcher from the container, we can call `::getRouteMatch()` which is supplied by the base class.